src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPrefooters.java

```diff
@@ -0,0 +1,34 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.dataprovider.ads.AdsDataProvider;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import org.testng.annotations.Test;
+
+/**
+ * @ownership AdEngineering Wikia
+ */
+public class TestAdsPrefooters extends TemplateNoFirstLoad {
+
+  @Test(
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsMiddlePrefooter",
+      groups = "AdsMiddlePrefooter"
+  )
+  public void adsMiddlePrefooter(String wikiName, String path, boolean middlePrefooterEnabled) {
+    String testedPage = urlBuilder.getUrlForPath(wikiName, path);
+    AdsBaseObject adsBaseObject = new AdsBaseObject(driver, testedPage);
+    adsBaseObject.scrollToFooter();
+
+    boolean middlePrefooterOnPage = adsBaseObject.isMiddlePrefooterOnPage();
+
+    Assertion.assertEquals(middlePrefooterOnPage, middlePrefooterEnabled,
+                           "Middle prefooter on page: " + middlePrefooterOnPage);
+
+    if (middlePrefooterEnabled) {
+      adsBaseObject.verifyMiddlePrefooterAdPresent();
```

Comment:
:+1: 
I thought about it as well but it requires more refactoring.
Let's start to use this convention after refactoring tests.